### PR TITLE
Fix a number of tests on Test WHS

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fixes a number of tests on Test WHS

## Tests
ReactOS: https://reactos.org/testman/compare.php?ids=85718,85719,85721,85724
Test WHS: ...

## TODO

- [ ] Test WHS run

### CODE REVIEW

File: modules/rostests/apitests/ntdll/Timer.c [hunk 1]
```diff
@@ -60,7 +60,9 @@ START_TEST(TimerResolution)
                                   &CurrentResolution);
     ok_hex(Status, STATUS_SUCCESS);
     printf("Current resolution: %lu ; maximum resolution: %lu\n", CurrentResolution, MaximumResolution);
+#if 0 // Why? This doesn't hold true on WIndows Server 2003.
```
Comment (katahiromz) [inline modules/rostests/apitests/ntdll/Timer.c:63]:
```suggestion
#if 0 // Why? This doesn't hold true on Windows Server 2003.
```

File: modules/rostests/apitests/ntdll/Timer.c [hunk 2]
```diff
@@ -60,7 +60,9 @@ START_TEST(TimerResolution)
...
     ok(abs((LONG)MaximumResolution - (LONG)CurrentResolution) < 200, "Current resolution: %lu became too low! (maximum resolution: %lu)\n", CurrentResolution, MaximumResolution);
```
Comment (katahiromz) [inline modules/rostests/apitests/ntdll/Timer.c:64]:
```suggestion
    ok(labs((LONG)MaximumResolution - (LONG)CurrentResolution) < 200, "Current resolution: %lu became too low! (maximum resolution: %lu)\n", CurrentResolution, MaximumResolution);
```

File: modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c [hunk 3]
```diff
@@ -82,20 +82,20 @@ Test_Params(void)
     TEST(ret == 0);
     TEST(GetLastError() == ERROR_INVALID_HANDLE);
 
-    SetLastError(0);
+    SetLastError(0xdeadbeef);
     ret = NtGdiPolyPolyDraw(hDC, Points, Count1, 2, 3);
     TEST(ret == 0);
-    TEST(GetLastError() == ERROR_INVALID_HANDLE);
+    ok_int(GetLastError(), 0xdeadbeef);
```
Comment (katahiromz) [inline modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c:88]:
```suggestion
    ok_long(GetLastError(), 0xdeadbeef);
```

File: modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c [hunk 4]
```diff
@@ -82,20 +82,20 @@ Test_Params(void)
...
 
-    SetLastError(0);
+    SetLastError(0xdeadbeef);
     ret = NtGdiPolyPolyDraw(hDC, Points, Count1, 2, 4);
     TEST(ret == 0);
-    TEST(GetLastError() == ERROR_INVALID_HANDLE);
+    ok_int(GetLastError(), 0xdeadbeef);
```
Comment (katahiromz) [inline modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c:93]:
```suggestion
    ok_long(GetLastError(), 0xdeadbeef);
```

File: modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c [hunk 5]
```diff
@@ -82,20 +82,20 @@ Test_Params(void)
...
 
-    SetLastError(0);
+    SetLastError(0xdeadbeef);
     ret = NtGdiPolyPolyDraw(hDC, Points, Count1, 2, 5);
     TEST(ret == 0);
-    TEST(GetLastError() == ERROR_INVALID_HANDLE);
+    ok_int(GetLastError(), 0xdeadbeef);
```
Comment (katahiromz) [inline modules/rostests/apitests/win32nt/ntgdi/NtGdiPolyPolyDraw.c:98]:
```suggestion
    ok_long(GetLastError(), 0xdeadbeef);
```

**Review by learn-more — APPROVED** [review]

File: modules/rostests/apitests/ntdll/Timer.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/apitests/ntdll/Timer.c:64]: Why? `abs ` works fine here.

Comment (katahiromz) [reply modules/rostests/apitests/ntdll/Timer.c:64]: You can ignore my suggestions.

**Review by GeoB99 — APPROVED** [review]

**Review by Extravert-ir — APPROVED** [review]

File: modules/rostests/apitests/compiler/ms_seh.c [hunk 6]
```diff
@@ -89,11 +89,15 @@ START_TEST(ms_seh)
     run_test(seh0006);
     run_test(seh0007);
     run_test(seh0008);
+#ifdef _USE_NATIVE_SEH
```
Comment (HBelusca) [inline modules/rostests/apitests/compiler/ms_seh.c:92]: I don't see this define defined anywhere in e.g. some cmake file or a header:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=_USE_NATIVE_SEH

File: modules/rostests/apitests/crt/strlen.c [hunk 7]
```diff
@@ -48,7 +48,12 @@ Test_strlen(PFN_STRLEN pstrlen)
     eflags = __readeflags();
     __writeeflags(eflags | EFLAGS_DF);
     len = pstrlen(teststr + 4);
-    ok((__readeflags() & EFLAGS_DF) != 0, "Direction flag in ELFAGS was changed.");
+
+#ifdef _M_AMD64
+    ok((__readeflags() & EFLAGS_DF) != 0, "Direction flag in ELFAGS was changed.\n");
```
Comment (HBelusca) [inline modules/rostests/apitests/crt/strlen.c:53]: The string should read "EFLAGS", not "ELFAGS" (and same in the other one below).

**Pull request merged**